bin/test-launcher

```diff
@@ -0,0 +1,181 @@
+
+    >>> run_cmd('ls file_i_hope_doesnt_exist')[0] != 0
+    True
+    """
+    arg_stderr = subprocess.STDOUT if combine_output else arg_stderr
+    from_dir = str(from_dir) if from_dir else from_dir
+
+    if verbose:
+        print("RUN: {}\nFROM: {}".format(cmd, os.getcwd() if from_dir is None else from_dir))
+
+    if dry_run:
+        return 0, "", ""
+
+    if input_str is not None:
+        stdin = subprocess.PIPE
+        input_str = input_str.encode('utf-8')
+    else:
+        stdin = None
+
+    proc = subprocess.Popen(cmd,
+                            shell=True,
+                            stdout=arg_stdout,
+                            stderr=arg_stderr,
+                            stdin=stdin,
+                            cwd=from_dir,
+                            env=env)
+
+    output, errput = proc.communicate(input_str)
+    if output is not None:
+        try:
+            output = output.decode('utf-8', errors='ignore')
+            output = output.strip()
+        except AttributeError:
+            print("Something funcky for output")
+            pass
+    if errput is not None:
+        try:
+            errput = errput.decode('utf-8', errors='ignore')
+            errput = errput.strip()
+        except AttributeError:
+            print("Something funcky for errput")
+            pass
+
+    stat = proc.wait()
+
+    return stat, output, errput
+
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+        usage="""\n{0} <executable>
+OR
+{0} --help
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# Lightweight python wrapper for a test that handles thread binding.\033[0m
+    > {0} ./my_exec
+
+""".format(pathlib.Path(args[0]).name),
+        description=description,
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    parser.add_argument("-e","--executable", help="The name of the executable to run",required=True)
+    #  parser.add_argument("exec_args",nargs=argparse.REMAINDER,help="Additional args to fwd to the executable")
+
+    return parser.parse_known_args(args[1:])
+
+###############################################################################
+def run_test(executable,exec_args):
+###############################################################################
+
+    print("execname: {}".format(executable))
+
+    omp_env = []
+    # Uncomment the next line if you want to print OMP env and bindings when the executable runs
+    # omp_env.append("OMP_DISPLAY_ENV=verbose")
+    # omp_env.append("OMP_DISPLAY_AFFINITY=true")
+    omp_env.append("OMP_PROC_BIND=spread")
+    omp_env.append("OMP_PLACES=threads")
+
+    env = os.environ
+
+    # Check if we're run with MPI
+    if "OMPI_COMM_WORLD_SIZE" in env:
+        nranks = int(env["OMPI_COMM_WORLD_SIZE"])
```

Comment:
is `OMPI_COMM_WORLD_SIZE` portable across mpi types or only applies to openmpi?

Reply:
Ah, good catch! I noticed that in Kokkos, and meant to fix it, but I forgot. I'll fix it.